runtime/parachains/src/runtime_api_impl/vstaging.rs

```diff
@@ -44,3 +45,93 @@ pub fn submit_unsigned_slashing_report<T: disputes::slashing::Config>(
 		key_ownership_proof,
 	)
 }
+
+/// Implementation for the `availability_cores_staging` function of the runtime API.
+pub fn availability_cores<T: initializer::Config>(
+) -> Vec<vstaging::CoreState<T::Hash, T::BlockNumber>> {
+	let cores = <scheduler::Pallet<T>>::availability_cores();
+	let config = <configuration::Pallet<T>>::config();
+	let now = <frame_system::Pallet<T>>::block_number() + One::one();
+	let rotation_info = <scheduler::Pallet<T>>::group_rotation_info(now);
+
+	let time_out_at = |backed_in_number, availability_period| {
+		let time_out_at = backed_in_number + availability_period;
+
+		let current_window = rotation_info.last_rotation_at() + availability_period;
+		let next_rotation = rotation_info.next_rotation_at();
+
+		// If we are within `period` blocks of rotation, timeouts are being checked
+		// actively. We could even time out this block.
+		if time_out_at < current_window {
+			time_out_at
+		} else if time_out_at <= next_rotation {
```

Comment:
This code is out of sync with the updated timeout predicate. It is bad that this can even happen, we should be using the same logic. DRY.

Reply:
Agreed! The issue is that the updated timeout predicate is in the scheduler itself. The runtime API uses its own implementation for reasons unknown to me. Will unify.